### PR TITLE
fix: waitForTransactionReceipt proper eth_call usage

### PR DESCRIPTION
**Pull request by AndriyAntonenko**
This PR addresses a bug in the @wagmi/core library, specifically in the waitForTransactionReceipt function. The bug occurs when arguments are provided to skip simulation and gas estimation. The data field was not being passed to the eth_call function, which resulted in unexpected errors.

I've made a simple repository that has a script that reproduces this issue: https://github.com/AndriyAntonenko/wagmi-waitfortransaction-bug-poc

### CODE REVIEW

File: packages/core/src/actions/waitForTransactionReceipt.ts [hunk 1]
```diff
@@ -65,17 +76,34 @@ export async function waitForTransactionReceipt<
...
+      if (
+        contract &&
+        error instanceof CallExecutionError &&
+        error.cause instanceof ExecutionRevertedError &&
+        error.cause.cause instanceof RpcRequestError
+      ) {
+        throw new ContractFunctionRevertedError({
+          abi: contract.abi as Abi,
+          functionName: contract.functionName,
+          data: (error.cause.cause.cause as any).data,
```
Comment (dalechyn) [inline packages/core/src/actions/waitForTransactionReceipt.ts:103]: can typing be slightly improved here?

**Comment by tmm** [comment]
Let's update this PR so it only fixes the issue and doesn't add new features.

> Also, I've added the possibility to pass abi and functionName as optional params in waitForTransactionReceipt. These arguments will be used to parse the possible error thrown by action_call function

Ideally this type of thing is added to Viem so Wagmi gets it for free. Please [create a discussion](https://github.com/wevm/viem/discussions) over there before creating a PR.

File: packages/core/src/actions/waitForTransactionReceipt.ts (hunk 1 above)
Comment (AndriyAntonenko) [reply packages/core/src/actions/waitForTransactionReceipt.ts]: Removed this code as suggested in commend below

**Comment by AndriyAntonenko** [comment]
@tmm thanks, I've removed the part with error handling. Now this PR only fix the bug

**Pull request merged**